cmd/kubeadm/app/phases/etcd/local.go

```diff
@@ -36,10 +39,47 @@ const (
 )
 
 // CreateLocalEtcdStaticPodManifestFile will write local etcd static pod manifest file.
+// This function is used by init (when there the etcd cluster is empty) or by kubeadm
+// upgrade (when the etcd cluster is already setup and initial cluster setting have no impacts)
 func CreateLocalEtcdStaticPodManifestFile(manifestDir string, cfg *kubeadmapi.InitConfiguration) error {
 	glog.V(1).Infoln("creating local etcd static pod manifest file")
 	// gets etcd StaticPodSpec, actualized for the current InitConfiguration
-	spec := GetEtcdPodSpec(cfg)
+	emptyInitialCluster := map[string]string{}
+	spec := GetEtcdPodSpec(cfg, emptyInitialCluster)
+	// writes etcd StaticPod to disk
+	if err := staticpodutil.WriteStaticPodToDisk(kubeadmconstants.Etcd, manifestDir, spec); err != nil {
+		return err
+	}
+
+	fmt.Printf("[etcd] Wrote Static Pod manifest for a local etcd instance to %q\n", kubeadmconstants.GetStaticPodFilepath(kubeadmconstants.Etcd, manifestDir))
+	return nil
+}
+
+// CreateStackedEtcdStaticPodManifestFile will write local etcd static pod manifest file
+// for an etcd member that is joining an existing etcd cluster.
+// Other members of the etcd cluster will be notified of the joining node in beforehand as well.
+func CreateStackedEtcdStaticPodManifestFile(client clientset.Interface, manifestDir string, cfg *kubeadmapi.InitConfiguration) error {
+
+	// creates an etcd client that connects to all the pods with label component=etcd
+	glog.V(1).Infoln("creating etcd client that connects to etcd pods")
+	etcdClient, err := etcdutil.NewFromPods(client, manifestDir, cfg.CertificatesDir)
+	if err != nil {
+		return err
+	}
+
+	// notifies the other members of the etcd cluster about the joining member
+	etcdPeerAddress := fmt.Sprintf("https://%s:2380", cfg.APIEndpoint.AdvertiseAddress)
+
+	glog.V(1).Infof("Adding etcd Member %s", etcdPeerAddress)
+	initialCluster, err := etcdClient.AddMember(cfg.NodeRegistration.Name, etcdPeerAddress)
+	if err != nil {
+		return err
+	}
+	glog.V(1).Infof("Updated etcd member list %v", initialCluster)
```

Comment:
We should do a local client member check, to verify it's correct, and to list the current members in the log line.